src/main/java/com/aws/greengrass/shadowmanager/sync/IotDataPlaneClientFactory.java

```diff
@@ -125,4 +143,33 @@ private void configureClient(DeviceConfiguration deviceConfiguration) {
         }
         this.iotDataPlaneClient = iotDataPlaneClientBuilder.build();
     }
+
+    /**
+     * Getter for IoT data plane client. This configures the client everytime the getter is used.
+     *
+     * @return iotDataPlaneClient
+     * @throws IoTDataPlaneClientCreationException exception during client configuration
+     */
+    public IotDataPlaneClient getIotDataPlaneClient() throws IoTDataPlaneClientCreationException {
+        configureClient();
```

Comment:
this isn't safe.

Configure sets the client in a field which you then return from. A different thread could run the callback which sets the client to null, now you just returned null to your caller.

Reply:
this still isn't safe. You're using atomic, but without any locking this is exactly equivalent in safety to what you had before.

Have configureClient return you a client and return it here to actually be safe.